### ISSUE TITLE
HID USB device not ready. hid returned 127.

### ISSUE THREAD

**Issue by lucasuait** [urbanadventurer/Android-PIN-Bruteforce#82]
I have OP6 with Lineage OS and Kali Nethunter Version: 2021.2(2021020200)
![photo_2021-07-08_20-30-02](https://user-images.githubusercontent.com/69118979/125002440-4cb3cf00-e02b-11eb-91d9-06725d9aab67.jpg)

**Comment by invincible-man** [comment]
Same issue is happening with me!! Didn't you find some fixes yet..?

**Comment by invincible-man** [comment]
> I have OP6 with Lineage OS and Kali Nethunter Version: 2021.2(2021020200)
> ![photo_2021-07-08_20-30-02](https://user-images.githubusercontent.com/69118979/125002440-4cb3cf00-e02b-11eb-91d9-06725d9aab67.jpg)

Same issue is happening with me!! Didn't you find some fixes yet..?

**Comment by urbanadventurer** [comment]
@lucasuait @invincible-man 

Does this problem consistently occur every time?

**Comment by urbanadventurer** [comment]
The $HID_KEYBOARD variable looks wrong.

**Comment by L4F33R** [comment]
Yes . In config default file contains above path. But not working ,only shows hid usb devic3 not ready

**Comment by invincible-man** [comment]
> @lucasuait @invincible-man
> 
> Does this problem consistently occur every time?
> 
> The $HID_KEYBOARD variable looks wrong. It should show a full path.
> Do you have a line in config.default that contains `HID_KEYBOARD=/system/xbin/hid-keyboard`?

Thanks for considering above issues, mine has finally worked after again cloning the script but this time on termux unlike the first times I was trying through kali nethunter and the error for usb is not ready I was seen there has gone. 
Can't express more for why but I think its an permission access issue to the files of system..!

**Comment by Krystianp00000** [comment]
![Uploading 16289356967214261345380043675589.jpg…]()

**Comment by Krystianp00000** [comment]
https://github.com/urbanadventurer/Android-PIN-Bruteforce/issues/82#issue-940292489

**Comment by zhegan2008** [comment]
> > @lucasuait @invincible-man
> > Does this problem consistently occur every time?
> > The $HID_KEYBOARD variable looks wrong. It should show a full path.
> > Do you have a line in config.default that contains `HID_KEYBOARD=/system/xbin/hid-keyboard`?
> 
> Thanks for considering above issues, mine has finally worked after again cloning the script but this time on termux unlike the first times I was trying through kali nethunter and the error for usb is not ready I was seen there has gone. Can't express more for why but I think its an permission access issue to the files of system..!

How did you clone script? Write, please, commands

**Comment by varsismaname** [comment]
I have error 126
[FAIL] HID USB device not ready.  /system/xbin/hid-keyboard returned 126.

**Comment by Sunilkumarchoudari** [comment]
[FAIL] HID device (/dev/hidg0) not found

**Comment by mebuis** [comment]
I have error 127 In Nethunter 2022.1  and the cell phone is oneplus7
[FAIL] HID USB device not ready. /system/xbin/hid-keyboard returned 127.

how to resolve it?

**Comment by edongab23** [comment]
me too, using OP7 Pro.
[FAIL] HID USB device not ready. /system/xbin/hid-keyboard returned 127.
any help please?

**Comment by abdoatman** [comment]
40 containers and 45 containers will always be booked from 3 to 4 containers per month of car loadمُرسلة من هاتفي-------- الرسالة الأصلية --------من:‏ edongab23 ***@***.***>التاريخ:‏ الخميس، ٢٩ ديسمبر، ٢٠٢٢ ١:٠٢ مإلى:‏ urbanadventurer/Android-PIN-Bruteforce ***@***.***>نسخة إلى:‏ Subscribed ***@***.***>الموضوع:‏ Re: [urbanadventurer/Android-PIN-Bruteforce] HID USB device not ready. hid returned 127. (#82)
me too, using OP7 Pro.
[FAIL] HID USB device not ready. /system/xbin/hid-keyboard returned 127.
any help please?

**Comment by IamCOD3X** [comment]
Here is the solution https://youtu.be/OoMctGTSQ-A

**Comment by Host-Nik** [comment]
This guide actually helped me
Hid-Keyboard binary fix: https://github.com/urbanadventurer/Android-PIN-Bruteforce/issues/120
